src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Helpers.cs

```diff
@@ -29,6 +29,14 @@ public sealed partial class ConfigurationBindingSourceGenerator
             defaultSeverity: DiagnosticSeverity.Info,
             isEnabledByDefault: true);
 
+        private static DiagnosticDescriptor LanguageVersionNotSupported { get; } = new DiagnosticDescriptor(
+            id: "SYSLIB1102",
+            title: new LocalizableResourceString(nameof(SR.LanguageVersionIsNotSupportedTitle), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+            messageFormat: new LocalizableResourceString(nameof(SR.Language_VersionIsNotSupportedMessageFormat), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+            category: GeneratorProjectName,
+            defaultSeverity: DiagnosticSeverity.Info,
```

Comment:
Does it make sense that this is `Info`? If someone explicitly enabled the source generator, and weren't able to use it because the LangVersion isn't set high enough, shouldn't that be a warning?

When I try using the Regex source generator with `<LangVersion>10</LangVersion>` in my project, I get a build error:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8936	Feature 'file types' is not available in C# 10.0. Please use language version 11.0 or greater.	ConsoleApp85	C:\Users\eerhardt\source\repos\ConsoleApp85\ConsoleApp85\System.Text.RegularExpressions.Generator\System.Text.RegularExpressions.Generator.RegexGenerator\RegexGenerator.g.cs	33	Active
```

Reply:
Perhaps an error is better given that we'd be falling back to reflection if the build succeeds & we don't generate anything. Then the resolution would be to either update the language version, or to remove the `<Enable...Generator/>` property from their project.

Switch to error in the latest commit.

Reply:
I agree error is better here.